src/Simple.OData.Client.Core/Extensions/DictionaryExtensions.cs

```diff
@@ -281,5 +287,21 @@ private static ODataEntry CreateODataEntry(IDictionary<string, object> source, I
             property.SetValue(instance, dynamicProperties, null);
             return dynamicProperties;
         }
+
+        private static object CreateInstanceOfAnonymousType(IDictionary<string, object> source, Type type, ITypeCache typeCache)
+        {
+            var constructorParameterTypes = source.Values.Select(v => v.GetType()).ToArray();
```

Comment:
I like this method, but it will be expensive on the hot path given three Select's and three ToArray's. All three collections could be exactly sized arrays as well.

It also relies on the fact that `Dictionary<K,V>` iterates in the order of insertion which is an implementation detail.

Reply:
I have refactored the creation of instances of anonymous type to avoid using `.Select(...).ToArray()` multiple times